### PR TITLE
Fixed issue #2461 -> OutOfMemory exception on opening big text files in editor

### LINKED ISSUE #2461: OutOfMemory exception on opening big text files in editor
## Exception
* User Action: UI Error
* Request: Application crash
* Version: 3.5.3
* OS: Linux samsung/crownltexx/crownlte:10/QP1A.190711.020/N960FXXU6FTK1:user/release-keys 10 - 29
* Device: crownlte
* Model: SM-N960F
* Product: crownltexx
<details><summary><b>Crash log </b></summary><p>

java.lang.RuntimeException: An error occurred while executing doInBackground()
 at android.os.AsyncTask$4.done(AsyncTask.java:399)
 at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
 at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
 at java.util.concurrent.FutureTask.run(FutureTask.java:271)
 at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
 at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.OutOfMemoryError: Failed to allocate a 174063624 byte allocation with 25149440 free bytes and 165MB until OOM, target footprint 119906520, growth limit 268435456
 at java.util.Arrays.copyOf(Arrays.java:3257)
 at java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:124)
 at java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:448)
 at java.lang.StringBuilder.append(StringBuilder.java:137)
 at com.amaze.filemanager.asynchronous.asynctasks.ReadFileTask.doInBackground(ReadFileTask.java:118)
 at com.amaze.filemanager.asynchronous.asynctasks.ReadFileTask.doInBackground(ReadFileTask.java:47)
 at android.os.AsyncTask$3.call(AsyncTask.java:378)
 at java.util.concurrent.FutureTask.run(FutureTask.java:266)
 ... 4 more

</details>
<hr>

### PR DESCRIPTION
**Pull request by VishnuSanal**
#### Issue tracker   

Fixes #2461 

#### Release  
Addresses release/3.6
  
#### Manual testing
- [x] Done  

#### Build tasks success  
Successfully running following tasks on local 
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

**Comment by VishalNehra** [comment]
Did you test this? Can you attach a file to test?

**Comment by VishnuSanal** [comment]
> Did you test this? Can you attach a file to test?

Nope!

**Comment by EmmanuelMess** [comment]
> > Did you test this? Can you attach a file to test?
> 
> Nope!

You can't expect things you didn't test to work.

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ReadFileTask.java [hunk 1]
```diff
@@ -126,6 +127,9 @@ protected ReturnedValues doInBackground(Void... params) {
     } catch (IOException e) {
       e.printStackTrace();
       return new ReturnedValues(EXCEPTION_IO);
+    } catch (OutOfMemoryError e) {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ReadFileTask.java:130]: As I commented on telegram, catching OOM has issues, like the fact that you don't actually have any assurance of being able to create `new` objects.

**Review by EmmanuelMess — CHANGES_REQUESTED** [review]
Please test and attach test.

**Pull request closed without merging**